### DISCUSSION TITLE
Color preview in Ghidra

### DISCUSSION THREAD

**Discussion by Nemoumbra** [NationalSecurityAgency/ghidra#6147] [Q&A]
I've seen that Ghidra can locate embedded media such as PNG images or WAV files in binaries and create special entities in the Listing view. Is there anything that can be used to preview the colors that look like that?
![image](https://github.com/NationalSecurityAgency/ghidra/assets/62447396/61c8822c-c0a5-49b3-b264-5ebc275188f5)

I expected to see a few builtin types such as RGBA, RGB565, etc.

There's also a `Data Preview Window`, but I think it can't be used to preview the colors.

**Comment by dragonmacher** [comment]
There is nothing currently in Ghidra that can do this.   This is probably fairly easy to do by adding a new Field Factory, which is how the `Listing` image preview is implemented.   It could also be done with a new component provider, which is how the `Data Preview Window` is implemented.
